### DISCUSSION TITLE
Uso de SVN en el framework

### DISCUSSION THREAD

**Discussion by tomasdelvechio** [SIU-Toba/framework#176] [Ideas]
Buenas, tenemos una consulta. Cada vez que se hace una petición a un proyecto toba, dentro de [la carga del contexto de ejecución](https://github.com/SIU-Toba/framework/blob/develop/php/nucleo/toba_nucleo.php#L331), en un momento intenta [recuperar un número de revisión](https://github.com/SIU-Toba/framework/blob/develop/php/nucleo/toba_nucleo.php#L339) (linea 339) fijándose si existe svn o es una working copy de un repo svn.

En la mayoría de los servers ya no está instalado svn y en muchos casos no son entornos de producción, con lo cual, se nos llenan los logs de mensajes como estos

```bash
[ERROR][proyecto] sh: 1: svn: not found

[ERROR][proyecto] sh: 1: svn: not found
```
La primera intención fue comentar la llamada al método `recuperar_revision_recursos` de la clase `toba_nucleo`. Pero nos dimos cuenta que cuando svn no está, setea 2 variables en sesión.

Las preguntas son:

 - Cual es la viabilidad de meter un if con una variable de entorno donde se pueda habilitar o deshabilitar svn, y en caso de que este desactivado, se usen valores por default de las variables en sesión???
 - En caso que lo anterior no se pueda por algo que no estamos viendo, suponiendo que tenemos una instalación que ya no es producción pero queremos ocultar ese mensaje, cual es la mejor manera???

**Comment by enfoqueNativo** [comment]
Hola @tomasdelvechio   
> La primera intención fue comentar la llamada al método `recuperar_revision_recursos` de la clase `toba_nucleo`. Pero nos dimos cuenta que cuando svn no está, setea 2 variables en sesión.

Las variables de sesion estan destinadas a agregar un parametro cambiante a la URL para que al momento de recuperar los recursos (sobre todo JS) el browser este forzado a refrescarlos.  Lo que pasaba es algunos browsers cacheaban los recursos JS (independientemente de lo que solicitaran los headers) y en los cambios de version no se refrescaban los mismos generando problemas para los clientes.

> Las preguntas son:
> 
> * Cual es la viabilidad de meter un if con una variable de entorno donde se pueda habilitar o deshabilitar svn, y en caso de que este desactivado, se usen valores por default de las variables en sesión???
> * En caso que lo anterior no se pueda por algo que no estamos viendo, suponiendo que tenemos una instalación que ya no es producción pero queremos ocultar ese mensaje, cual es la mejor manera???

Lo ideal creo seria tener una env-var que defina el control de versiones a usar y la correspondiente clase que implemente una interface comun.  En el caso base, la clase responde siempre que no hay cliente y nos vamos derecho viejo por el parseo del archivo de version.
Luego si la env-var no existe o no tiene valor, se usa dicha clase base. Que te parece algo asi?.. el nombre de la env decidilo porque no se me ocurre nada :smile: 

Como workaround inmediato (para no generar polucion en el log)  pueden redefinir el metodo `recuperar_revision_recursos`  (temporal hasta que meta el cambio)  y fijar las variables de sesion con las lineas que dicen `get_version()->__toString();`

**Reply by tomasdelvechio** [reply]
Si quiero levantar el framework para desarrollar sobre el, cual es la forma adecuada?

Con este proyecto forkeado, intente hacer un `docker compose up` y me doy cuenta que el archivo `toba.sh` no existe mas, entonces no me instaló nada.

Tengo que levantar el container y luego ingresar e instalar "a la antigua"??? o hay otra forma recomendada de instalarlo???

**Reply by enfoqueNativo** [reply]
> Tengo que levantar el container y luego ingresar e instalar "a la antigua"??? o hay otra forma recomendada de instalarlo???

This.. una vez levantaste los contenedores vuelve "la vieja guardia" ya que de todas maneras tenes que:

- Instalar composer / yarn
- Lanzar los comandos adecuados
- Enlazar el toba.conf o copiarlo y  stop/start del contenedor
